### PR TITLE
Remove go compiler install on CI

### PR DESCRIPTION
**Pull request by aslonnie**
The only place using golang compiler is the buildifier, and we can just download released buildifier binary rather than building from source.

### CODE REVIEW

File: ci/ci.sh [hunk 1]
```diff
@@ -543,17 +533,22 @@ lint_annotations() {
 }
 
 lint_bazel() {
-  # Run buildifier without affecting external environment variables
-  (
-    mkdir -p -- "${GOPATH}"
-    export PATH="${GOPATH}/bin:${GOROOT}/bin:${PATH}"
+  if [[ ! "${OSTYPE}" =~ ^linux ]]; then
+    echo "Bazel lint not supported on non-linux systems."
+    exit 1
```
Comment (can-anyscale) [inline ci/ci.sh:538]: is this function currently called on non-supported system?

**Review by can-anyscale — APPROVED** [review]

File: ci/ci.sh (hunk 1 above)
Comment (aslonnie) [reply ci/ci.sh:538]: I searched and no. this is only called on ci, and ci only runs lints on linux `x86_64`.

People can still run the bazel lint directly, which assumes that `buildifier` is already installed on the system.

**Comment by aslonnie** [comment]
lint passing: https://buildkite.com/ray-project/oss-ci-build-pr/builds/25409#0188b6fc-b20d-44b0-8d43-1a29a381df1d

**Review by matthewdeng — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/viceland/">VICELAND</a> on <a href="https://giphy.com/gifs/viceland-xT0BKJGfxmloVAqjss">GIPHY</a></div>

**Pull request merged**